### DISCUSSION TITLE
telemetry with on-prem deployment

### DISCUSSION THREAD

**Discussion by maxkomarychev** [wundergraph/cosmo#680] [Q&A] [answered]
Is there a way to set up components of the app (specifically router, controlplane) to appear in traces if the app is deployed with already existing OTEL-compatible collector? (datadog for example like in this case: https://github.com/wundergraph/cosmo/issues/90)

Use case: I want to be able to see traces and spans produced by the router outside of the cosmo dashboard.

Right now I see spans from my subgraphs but they are "floating in the air" without connection or understanding of what was the part of the router in query planning.

**Comment by StarpTech** [comment, marked as answer]
Yes, you can add your custom exporter in the `router.yml` file to export to different OTEL providers.
https://cosmo-docs.wundergraph.com/router/open-telemetry#multiple-exporters
